pkg/db/drivers/etcd/etcd.go

```diff
@@ -862,6 +1517,113 @@ func (c *Client) ListVolumeSnapshots() ([]*model.VolumeSnapshotSpec, error) {
 	return vss, nil
 }
 
+var volumeSnapshot_sortKey string
+
+type VolumeSnapshotSlice []*model.VolumeSnapshotSpec
+
+func (volumeSnapshot VolumeSnapshotSlice) Len() int { return len(volumeSnapshot) }
+
+func (volumeSnapshot VolumeSnapshotSlice) Swap(i, j int) {
+
+	volumeSnapshot[i], volumeSnapshot[j] = volumeSnapshot[j], volumeSnapshot[i]
+}
+
+func (volumeSnapshot VolumeSnapshotSlice) Less(i, j int) bool {
+	switch volumeSnapshot_sortKey {
+
+	case "ID":
+		return volumeSnapshot[i].Id < volumeSnapshot[j].Id
+
+	case "VOLUMEID":
+		return volumeSnapshot[i].VolumeId < volumeSnapshot[j].VolumeId
+
+	case "STATUS":
+		return volumeSnapshot[i].Status < volumeSnapshot[j].Status
+
+	case "USERID":
+		return volumeSnapshot[i].UserId < volumeSnapshot[j].UserId
+
+	case "PROJECTID":
+		return volumeSnapshot[i].ProjectId < volumeSnapshot[j].ProjectId
+	}
```

Comment:
VolumeSize?

Reply:
Add a TODO note for GroupSnapshotId.

Reply:
All ProjectId  will be modified by another PR